### DISCUSSION TITLE
Resize attached volume & extend Linux file system

### DISCUSSION THREAD

**Discussion by jihoon-seo** [cloud-barista/cb-tumblebug#457] [General]
CC @powerkimhub 

[개요]
- 예를 들어, CB-Spider + CB-Tumblebug 을 이용하여 AWS VM 을 생성하면,
기본적으로 함께 생성되는 EBS 볼륨의 크기는 8 GB 입니다.
- 이 EBS 볼륨과 리눅스 VM 파일 시스템을 확장하는 방법입니다.

[Resize attached volume]
- [탄력적 볼륨을 사용하여 EBS 볼륨 수정(콘솔)](https://docs.aws.amazon.com/ko_kr/AWSEC2/latest/UserGuide/requesting-ebs-volume-modifications.html)

> 1. https://console.aws.amazon.com/ec2/에서 Amazon EC2 콘솔을 엽니다.
> 
> 2. 볼륨을 선택하고, 수정할 볼륨을 선택한 다음, 작업, 볼륨 수정을 선택합니다.
> 
> 3. [볼륨 수정(Modify Volume)] 창에 볼륨 ID와 유형, 크기, IOPS 및 처리량을 포함한 볼륨의 현재 구성이 표시됩니다. 다음과 같이 새로운 구성 값을 설정합니다.
> 
>    - 유형을 수정하려면 볼륨 유형의 값을 선택합니다.
> 
>    - 크기를 수정하려면 [크기(Size)]에 대한 새 값을 입력합니다.
> 
>    - IOPS를 수정하려면 볼륨 유형이 gp3, io1 또는 io2인 경우 [IOPS]에 대한 새 값을 입력합니다.
> 
>    - 처리량을 수정하려면 볼륨 유형이 gp3인 경우 [처리량(Throughput)]에 대한 새 값을 입력합니다.
> 
> 4. 볼륨 설정 변경을 완료했으면 수정을 선택합니다. 확인 메시지가 표시되면 예를 선택합니다.
> 
> 5. 볼륨 크기를 수정해도 새 스토리지 용량을 활용하기 위해 볼륨의 파일 시스템도 확장할 때까지는 실질적인 효과가 없습니다. 자세한 내용은 볼륨 크기 조정 후 Linux 파일 시스템 확장 섹션을 참조하십시오.

[Extend Linux file system]
- [볼륨 크기 조정 후 Linux 파일 시스템 확장](https://docs.aws.amazon.com/ko_kr/AWSEC2/latest/UserGuide/recognize-expanded-volume-linux.html)

[예시]
```
$ df -hT

$ lsblk

$ sudo growpart /dev/xvda 1

$ lsblk

$ df -h

$ sudo resize2fs /dev/xvda1

$ df -h
```

**Comment by powerkimhub** [comment]
@jihoon-seo @seokho-son 

**<VM 생성시 변경>**
- VM 생성 요청시 Volume 크기 설정은 고민해보도록하겠습니다. 
- 모든(또는 다수의) CSP가 수정 가능한지 등 선행 확인 작업이 필요해 보입니다. 

**<VM 운영시 변경>**
- Volume 추가 및 resize 등은 이종 Storage Service들의 통합 서비스에 가까워 보입니다. 
- 현재는 수용하기 힘들 듯 합니다. 기회가 된다면 다음 단계에...^^

**Reply by powerkimhub** [reply]
- cf) https://github.com/cloud-barista/cb-spider/issues/348
